### PR TITLE
Fix Iran time zone test case

### LINKED ISSUE #83901: Platform_TimeZoneNames failing for Iran time zones on macOS
## Build Information
Build: https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=216592
Build error leg or test failing: System.Tests.TimeZoneInfoTests.Platform_TimeZoneNames
Pull request: https://github.com/dotnet/runtime/pull/83755

## Error Message
Fill the error message using [known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssues.md#how-to-fill-out-a-known-issue-error-section).

```json
{
  "ErrorMessage": "(UTC+03:30) Iran Standard Time (Tehran)",
  "BuildRetry": false,
  "ErrorPattern": "",
  "ExcludeConsoleLog": true
}
```

#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|0|0|

### PR DESCRIPTION
**Pull request by tarekgh**
Fixes https://github.com/dotnet/runtime/issues/83901

In the recent ICU updates, there has been some changing in the Iran time zone display name which broke our test. The fix here is allow both the old and new names to ensure the test succeed in all platforms.

### CODE REVIEW

File: src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs [hunk 1]
```diff
@@ -141,8 +138,10 @@ public static void Platform_TimeZoneNames(TimeZoneInfo tzi, string displayName,
                 }
             }
 
-            Assert.Equal($"DisplayName: \"{displayName}\", StandardName: {standardName}\", DaylightName: {daylightName}\"",
-                         $"DisplayName: \"{tzi.DisplayName}\", StandardName: {tzi.StandardName}\", DaylightName: {tzi.DaylightName}\"");
+            Assert.True(displayName == tzi.DisplayName || alternativeDisplayName == tzi.DisplayName,
+                         $"Display Name: '{displayName}' not equal to '{tzi.DisplayName}'");
```
Comment (stephentoub) [inline src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs:142]:
```suggestion
                         $"Display Name: Neither '{displayName}' nor '{alternativeDisplayName}' equal to '{tzi.DisplayName}'");
```
?

**Review by stephentoub — APPROVED** [review]

**Comment by jkotas** [comment]
Test failures related to the change: `System.InvalidOperationException : The test method expected 5 parameter values, but 4 parameter values were provided.`

**Pull request merged**